psi4/src/psi4/libsapt_solver/exch-ind30.cc

```diff
@@ -314,5 +318,264 @@ double SAPT2p3::exch_ind30_3(double **sBS) {
 
     return (energy);
 }
+
+// Compute the nonapproximated third-order exchange-induction //
+// Konrad Patkowski, based on Jonathan Waldrop's Psi4NumPy code //
+// September 2021 //
+void SAPT2p3::sinf_e30ind() {
+
+    if (print_) {
+        outfile->Printf("  ==> Nonapproximated third-order induction <==\n\n");
+    }
+
+    // => Sizing <= //
+
+    int nn = basisset_->nbf();
+    int na = noccA_;
+    int nb = noccB_;
+    int nr = nvirA_;
+    int ns = nvirB_;
+    int nQ = ribasis_->nbf();
+
+    int nT = 1;
+#ifdef _OPENMP
+    nT = Process::environment.get_n_threads();
+#endif
+
+    auto uAR = std::make_shared<Matrix>("Ind30 uAR Amplitudes", na, nr);
+    auto uBS = std::make_shared<Matrix>("Ind30 uBS Amplitudes", nb, ns);
```

Comment:
I'd prefer `Matrix` over `shared_ptr<Matrix>`, unless there's a reason we need the pointer.

Reply:
Then I need to reveal my ignorance and ask a basic question: how to do this so that `uAR->load()` and `uAR->vector_dot()` still work?

Reply:
Just change `->` to `.`.

`->` is the syntax for "this object is not a Matrix but a memory address of a Matrix, so find the Matrix at this memory address and then call this Matrix method". `.` is the syntax for "this object is a Matrix, so call this Matrix method".

Reply:
OK, done, thanks!